### PR TITLE
Optimism and Arbitrum NFT support

### PR DESCRIPTION
**Pull request by jinchung**
## What changed (plus any additional context for devs)
This adds support for optimism and arbitrum NFTs.

The commit history is clean and is meant to be read in order.
Many items were tightly coupled with OpenSea, so many of the changes are nominal (name changes) or are updated to help the logic behave more generically.

## What to test

Example wallets used: vitalik.eth, misfitinternet.eth, 0x6F829Fd52DF9B664DDa4DA2D1ce200A404Ae0B80

Please test NFTs on mainnet, polygon, optimism, and arbitrum.
Please make sure our POAPs and ENS still work as expected
Please go through each item in the NFT expanded state.

Details about the expanded state:
* Rainbow web URLs are supported only for mainnet and polygon.
* The Share button should be the Rainbow web URL for mainnet and polygon; it should be the permalink for arbitrum and optimism when available. If the permalink is unavailable, the Share button should not be visible.
* The last sale and floor price section should not be visible for the L2s
* The block explorers are: Etherscan for mainnet and optimism; polygonscan and arbiscan
* The NFT marketplaces are: OpenSea for mainnet and polygon; Quixotic for optimism when available; Stratos for Arbitrum when available
* Please note that sometimes there is no marketplace link for the optimism and arbitrum NFTs
* There L2 disclaimers should be visible for any NFT on an L2
* When available, opening the NFT properties should direct you to the respective NFT marketplace
* Sending is currently unavailable for the L2s

Other changes:
* When tapping on the attributes, you should see "View All with Property" (with a lowercase "with" - previously it was "With")
* When you delete the app completely / clear cache and open a wallet for the first time, the NFT families should no longer load in batches

## Final checklist

- [ ] Assigned individual reviewers?
- [ ] Added labels? (`team1/team2`, `critical path`, `release`, `dev QA`)
- [ ] Did you test both iOS and Android?
- [ ] If your changes are visual, did you check both the light and dark themes?
- [ ] Added e2e tests? If not, please specify why
- [ ] If you added new critical path files, did you update the CODEOWNERS file?
- [ ] If no `dev QA` label, did you add the PR to the QA Queue?

### CODE REVIEW

File: src/parsers/uniqueTokens.js [hunk 1]
```diff
@@ -284,3 +284,58 @@ export const dedupeAssetsWithFamilies = (accountAssets, families) =>
...
+      image_thumbnail_url: lowResUrl,
+      image_url: imageUrl,
+      isPoap: false,
+      isSendable: false,
+      lastPrice: parseLastSalePrice(simplehashNft.last_sale?.unit_price),
+      lastSalePaymentToken: simplehashNft.last_sale?.payment_token?.symbol,
+      lowResUrl,
+      name: simplehashNft.name,
+      network: simplehashNft.chain,
+      permalink: null, // TODO - simplehashNft.collection.marketplace_pages / token_id
```
Comment (jinchung) [inline src/parsers/uniqueTokens.js]: when you see TODO's, they are fixed in a later commit

File: src/handlers/simplehash.ts [hunk 2]
```diff
@@ -39,3 +42,26 @@ export async function getNFTByTokenId({
     captureException(error);
   }
 }
+
+export async function getNftsByWalletAddress(walletAddress: string) {
```
Comment (jinchung) [inline src/handlers/simplehash.ts:101]: this commit is updated to use the cursor in a later commit

File: src/handlers/simplehash.ts [hunk 3]
```diff
@@ -2,12 +2,70 @@ import { captureException } from '@sentry/react-native';
...
+}
+
 const chains = {
+  arbitrum: 'arbitrum',
   ethereum: 'ethereum',
+  optimism: 'optimism',
+  polygon: 'polygon',
 } as const;
-type Chain = keyof typeof chains;
+type SimpleHashChain = keyof typeof chains;
```
Comment (jinchung) [inline src/handlers/simplehash.ts:66]: TODO JIN: update chain to Network

File: src/components/unique-token/Tag.js [hunk 4]
```diff
@@ -16,17 +16,21 @@ const HairlineSpace = '\u200a';
...
 
-const viewTraitOnOpenseaAction = {
-  actionKey: PropertyActionsEnum.viewTraitOnOpensea,
-  actionTitle: lang.t('expanded_state.unique_expanded.view_all_with_property'),
-  discoverabilityTitle: 'OpenSea',
-  icon: {
-    iconType: 'SYSTEM',
-    iconValue: 'magnifyingglass.circle.fill',
-  },
+const getViewTraitOnNftMarketplaceAction = (marketplaceName = 'OpenSea') => {
```
Comment (jinchung) [inline src/components/unique-token/Tag.js:22]: can now remove marketplaceName default of OpenSea as it is now included in parsers explicitly

File: src/components/unique-token/UniqueTokenAttributes.tsx [hunk 5]
```diff
@@ -10,15 +10,17 @@ import uniqueAssetTraitDisplayTypeCompareFunction from '@rainbow-me/helpers/uniq
 
 interface UniqueTokenAttributesProps {
   color: string;
+  hideNftMarketplaceAction?: boolean;
```
Comment (jinchung) [inline src/components/unique-token/UniqueTokenAttributes.tsx:13]: does not have to be ? optional - will have value

File: src/components/unique-token/UniqueTokenAttributes.tsx [hunk 6]
```diff
@@ -10,15 +10,17 @@ import uniqueAssetTraitDisplayTypeCompareFunction from '@rainbow-me/helpers/uniq
...
+  marketplaceName?: string;
```
Comment (jinchung) [inline src/components/unique-token/UniqueTokenAttributes.tsx:14]: check if this is optional vs null

File: src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx [hunk 7]
```diff
@@ -31,11 +42,20 @@ export default function NFTBriefTokenInfoRow({
...
   const [floorPrice, setFloorPrice] = useState<string | null>(null);
+
   useEffect(() => {
-    assetNetwork !== assetTypes.polygon &&
-      apiGetUniqueTokenFloorPrice(network, urlSuffixForAsset).then(result => {
-        setFloorPrice(result);
-      });
+    const isFloorPriceSupported = getIsFloorPriceSupported(assetNetwork);
+    if (isFloorPriceSupported) {
+      apiGetUniqueTokenFloorPrice(network, urlSuffixForAsset)
```
Comment (jinchung) [inline src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx:49]: move over to try catch syntax

**Comment by BrodyHughes** [comment]
If you're quickly looking for specific NFTs for test cases:

Arbitrum: 0x6F829Fd52DF9B664DDa4DA2D1ce200A404Ae0B80 - 6th anniversay NFT / Arbitrum Odyssey NFT
Optimism: Vitalik.eth - .op (Punk Domains)
Polygon: misfitinternet.eth - The Visitors
POAP: rainbowwallet.eth - POAP

**Comment by benjalimm** [comment]
@jinchung Do you think there will be support for Optimism/Arbitrum NFT video media? Currently, if an NFT has an mp4 video in the animation_url, it just shows up as blank on Rainbow. Thanks so much!

**Pull request merged**